### ISSUE TITLE
Can't connect to https://localhost:8080

### ISSUE THREAD

**Issue by f1nr0d** [searx/searx-docker#15]
I apologize if this is not the right forum to ask this question.

I am trying to install searx-docker on my MacBook Pro with macOS Mojave 10.14.6.

1. Installed Docker Desktop for Mac. It comes with docker-compose.
2. Edited `.env` to reflect:
    a. my real email address for let's encrypt
    b. Lets Encrypt TOS to `true`
    c. proper key for MORTY_KEY using the suggested command.
    d. password for filtron
3. Tried running start.sh with following output:
```
~/searx-docker   master ●  ./start.sh                                                                                                     1 ↵  163  13:15:33
readlink: illegal option -- f
usage: readlink [-n] [file ...]
use docker-compose.yaml
Removing network searx-docker_searx
Removing volume searx-docker_searx-checker
No stopped containers
Creating network "searx-docker_searx" with the default driver
Creating volume "searx-docker_searx-checker" with default driver
Pulling filtron (dalf/filtron:)...
latest: Pulling from dalf/filtron
921b31ab772b: Pull complete
c7ab82ae8f5a: Pull complete
098f07700af6: Pull complete
58e6837c81f2: Pull complete
Digest: sha256:9feef1cf833f11327f082c704de70f2a706672e1c23b2f90b9bb705d467f65dd
Status: Downloaded newer image for dalf/filtron:latest
Pulling searx (searx/searx:latest)...
latest: Pulling from searx/searx
89d9c30c1d48: Pull complete
1e2bb55f9cd3: Pull complete
826050ce83f2: Pull complete
2fff5bc55ea8: Pull complete
af5f5f249db4: Pull complete
ab4ae1b0f738: Pull complete
9ec27c60dc38: Pull complete
Digest: sha256:b08e8d371db8bd766fca453ba9f3b7c8967933ad993c08e974cbf58407a400c3
Status: Downloaded newer image for searx/searx:latest
Pulling morty (dalf/morty:)...
latest: Pulling from dalf/morty
9d48c3bd43c5: Pull complete
231de45363ec: Pull complete
52775b790275: Pull complete
Digest: sha256:7890614b280dbe0dcbd7a209ebe87989e80599f3627f37e1e0ca31f82e3a452e
Status: Downloaded newer image for dalf/morty:latest
Pulling searx-checker (searx/searx-checker:)...
latest: Pulling from searx/searx-checker
050382585609: Already exists
dac2222ca532: Pull complete
29a7fe408caa: Pull complete
6ad337b9b53f: Pull complete
31d663a76478: Pull complete
16d4fbd4b09b: Pull complete
ca41695224f1: Pull complete
eae20061edf5: Pull complete
30f46cf9aa58: Pull complete
3b28fb26a4c2: Pull complete
Digest: sha256:8548e831d672335a7301da966c1953bec12fff9084ad5b5a2426190dbca17300
Status: Downloaded newer image for searx/searx-checker:latest
Creating searx         ... done
Creating searx-checker ... done
Creating filtron       ... done
Creating caddy         ... done
Creating morty         ... done
Attaching to caddy, searx-checker, morty, searx, filtron
morty            | WARNING: no logs are available with the 'none' log driver
caddy            | 2019/11/19 21:17:37 [INFO][cache:0xc0001da370] Started certificate maintenance routine
searx-checker    | Sleep 84 seconds
caddy            | 2019/11/19 21:17:37 [WARNING] Stapling OCSP: no OCSP stapling for [localhost]: no OCSP server specified in certificate
caddy            | Activating privacy features... done.
filtron          | 2019/11/19 21:17:38 13 rules loaded from /etc/filtron/rules.json
filtron          | 2019/11/19 21:17:38 API listens on 10.10.10.3:4041
caddy            |
caddy            | Serving HTTPS on port 443
caddy            | https://localhost
caddy            |
caddy            | 2019/11/19 21:17:37 [INFO] Serving https://localhost
filtron          | 2019/11/19 21:17:38 Proxy listens on 10.10.10.3:4040
searx            | searx version 0.15.0-186-42d5e2c0
searx            |
searx            | Use existing /etc/searx/uwsgi.ini
searx            | Use existing /etc/searx/settings.yml
searx            | Listen on 10.10.10.4:8080
searx            | [uWSGI] getting INI configuration from /etc/searx/uwsgi.ini
searx-checker    | 2019/11/19 21:19:01 Check with the parameters : -o html/data/status.json http://10.10.10.4:8080
searx-checker    | Error get  /config , status code= 500
searx-checker    | Sleep 86315 seconds
```

I tried `https://localhost/`, `https://localhost:8080` in my browser but it doesn't connect.

**Comment by dalf** [comment]
If localhost is the only entry point (neither private nor public IP), you can keep the ```.env``` untouched.

```LETSENCRYPT_EMAIL``` and ```LETSENCRYPT_AGREE``` are used when you want to provide access to the instance outside of localhost.

**Comment by f1nr0d** [comment]
If I leave `.env` untouched then I get `Morty exited with code 1` errors every few seconds. If I put proper Morty key and that's the only thing changed in `.env`, then the same thing happens as I mentioned above.

**Comment by jfdhuiz** [comment]
on MacOS you want to use greadlink (GNU Readlink) instead of readlink

**Comment by f1nr0d** [comment]
I installed `greadlink` via `brew install coreutils`. But I still get the same error as above.
```
# ./start.sh                                                                                                                           ✔  271  13:51:27
use docker-compose.yaml
Removing filtron       ... done
Removing searx         ... done
Removing searx-checker ... done
Removing morty         ... done
Removing caddy         ... done
Removing network searx-docker_searx
Removing volume searx-docker_searx-checker
No stopped containers
Creating network "searx-docker_searx" with the default driver
Creating volume "searx-docker_searx-checker" with default driver
Creating filtron       ... done
Creating caddy         ... done
Creating searx-checker ... done
Creating searx         ... done
Creating morty         ... done
Attaching to caddy, searx, morty, searx-checker, filtron
caddy            | 2020/02/25 22:01:58 [INFO][cache:0xc0001a8370] Started certificate maintenance routine
caddy            | 2020/02/25 22:01:58 [WARNING] Stapling OCSP: no OCSP stapling for [localhost]: no OCSP server specified in certificate
caddy            | Activating privacy features... done.
caddy            |
caddy            | Serving HTTPS on port 443
caddy            | https://localhost
caddy            |
caddy            | 2020/02/25 22:01:58 [INFO] Serving https://localhost
filtron          | 2020/02/25 22:01:58 13 rules loaded from /etc/filtron/rules.json
filtron          | 2020/02/25 22:01:58 API listens on 10.10.10.3:4041
filtron          | 2020/02/25 22:01:58 Proxy listens on 10.10.10.3:4040
morty            | WARNING: no logs are available with the 'none' log driver
searx-checker    | Sleep 99 seconds
filtron          | 2020/02/25 22:01:58 Response error: dial tcp4 10.10.10.4:8080: connect: connection refused HTTP/1.1 200 OK
filtron          | Date: Tue, 25 Feb 2020 22:01:58 GMT
filtron          | Content-Length: 0
filtron          |
filtron          |
searx            | searx version 0.15.0-186-42d5e2c0
searx            |
searx            | Use existing /etc/searx/uwsgi.ini
searx            | Use existing /etc/searx/settings.yml
searx            | Listen on 10.10.10.4:8080
searx            | [uWSGI] getting INI configuration from /etc/searx/uwsgi.ini
searx-checker    | 2020/02/25 22:03:37 Check with the parameters : -o html/data/status.json http://10.10.10.4:8080
searx-checker    | Error get  /config , status code= 500
searx-checker    | Sleep 86301 seconds
```

**Comment by dalf** [comment]
The searx docker image doesn't start for some reason.

Can you give me the output:
```
docker exec -ti searx cat /var/log/uwsgi/uwsgi.log
```

It contains the uwsgi errors ( you should have this file : [./searx/uwsgi.ini](https://github.com/asciimoo/searx/blob/master/dockerfiles/uwsgi.ini#L26-L33) )

**Comment by sreme** [comment]
I am having the same issue.
I checked and my uwsgi.ini is the same as your link.
```
sudo docker exec -ti searx cat /var/log/uwsgi/uwsgi.log
*** Starting uWSGI 2.0.18 (64bit) on [Sun Jun  7 14:54:23 2020] ***
compiled with version: 8.3.0 on 13 November 2019 22:04:37
os: Linux-5.4.0-7629-generic #33~1589834512~20.04~ff6e79e-Ubuntu SMP Mon May 18 23:29:32 UTC 
nodename: fd688637c2cc
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 6
current working directory: /usr/local/searx
detected binary path: /usr/sbin/uwsgi
chdir() to /usr/local/searx/searx/
your memory page size is 4096 bytes
detected max file descriptor number: 1048576
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 0.0.0.0:8080 fd 6
Python version: 3.7.7 (default, May 30 2020, 09:56:15)  [GCC 8.3.0]
Python main interpreter initialized at 0x55b1877c5680
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
[uwsgi-cron] command "{ touch /run/uwsgi-logrotate  }" registered as cron task
mapped 364600 bytes (356 KB) for 4 cores
*** Operational MODE: preforking ***
added /usr/local/searx/ to pythonpath.
spawned uWSGI master process (pid: 6)
spawned uWSGI worker 1 (pid: 10, cores: 1)
spawned uWSGI worker 2 (pid: 11, cores: 1)
spawned uWSGI worker 3 (pid: 12, cores: 1)
spawned uWSGI worker 4 (pid: 13, cores: 1)
WSGI app 0 (mountpoint='') ready in 4 seconds on interpreter 0x55b1877c5680 pid: 11 (default app)
WSGI app 0 (mountpoint='') ready in 4 seconds on interpreter 0x55b1877c5680 pid: 12 (default app)
WSGI app 0 (mountpoint='') ready in 4 seconds on interpreter 0x55b1877c5680 pid: 13 (default app)
WSGI app 0 (mountpoint='') ready in 4 seconds on interpreter 0x55b1877c5680 pid: 10 (default app)
WARNING:searx.search:engine timeout: 1x
ERROR:searx.search:engine 1x : HTTP requests timeout(search duration : 2.1728789806365967 s, timeout: 2.0 s) : ReadTimeout
WARNING:searx.search:engine timeout: 1x
ERROR:searx.search:engine 1x : exception : Unicode strings with encoding declaration are not supported. Please use bytes input or XML fragments without declaration.
Traceback (most recent call last):
  File "/usr/local/searx/searx/search.py", line 156, in search_one_http_request_safe
    search_results = search_one_http_request(engine, query, request_params)
  File "/usr/local/searx/searx/search.py", line 97, in search_one_http_request
    return engine.response(response)
  File "/usr/local/searx/searx/engines/www1x.py", line 37, in response
    dom = html.fromstring(resp.text)
  File "/usr/lib/python3.7/site-packages/lxml/html/__init__.py", line 875, in fromstring
    doc = document_fromstring(html, parser=parser, base_url=base_url, **kw)
  File "/usr/lib/python3.7/site-packages/lxml/html/__init__.py", line 761, in document_fromstring
    value = etree.fromstring(html, parser, **kw)
  File "src/lxml/etree.pyx", line 3235, in lxml.etree.fromstring
  File "src/lxml/parser.pxi", line 1871, in lxml.etree._parseMemoryDocument
ValueError: Unicode strings with encoding declaration are not supported. Please use bytes input or XML fragments without declaration.
WARNING:searx.search:engine timeout: 1x
ERROR:searx.search:engine 1x : HTTP requests timeout(search duration : 2.0946409702301025 s, timeout: 2.0 s) : ReadTimeout
WARNING:searx.search:engine timeout: ccc-tv
ERROR:searx.search:engine ccc-tv : HTTP requests timeout(search duration : 3.215742349624634 s, timeout: 2.0 s) : Timeout
WARNING:searx.search:engine timeout: deviantart
ERROR:searx.search:engine deviantart : HTTP requests timeout(search duration : 3.210198402404785 s, timeout: 3.0 s) : ReadTimeout
```

**Comment by sreme** [comment]
I had this issue. Turns out you have to connect to just this: `https://localhost/`.

**Comment by 0hlov3** [comment]
I've the same issues...

Docker version 19.03.12, build 48a66213fe
docker-compose version 1.25.4, build 8d51620a

just edited the 'MORTY_KEY' to the right one. and sears will not start. 

[debug.log](https://github.com/searx/searx-docker/files/5099137/debug.log)

**Comment by f1nr0d** [comment]
Sorry, I should have closed this issue some time ago.

1. I essentially had to set permissions for /usr/local/searx-docker to 777.
2. Install greadlink as suggested above. It's not a standalone package though. Had to use this `brew install coreutils`
3. In `.env` file, changed the MORTY_KEY to proper value using `openssl rand -base64 33`
4. In `start.sh` file, changed `readlink` to `greadlink`
5. In `docker-compose.yaml` file, exposed the port 8080 for searx config. Not sure why this was needed though. Tried and it worked, so didn't pay much attention.
```searx:
    container_name: searx
    image: searx/searx:latest
    hostname: searx
    restart: always
    networks:
      searx:
         ipv4_address: 10.10.10.4
    command: ${SEARX_COMMAND:-}
    volumes:
      - ./searx:/etc/searx:rw
    environment:
      - BIND_ADDRESS=10.10.10.4:8080
      - BASE_URL=https://${SEARX_HOSTNAME}/
      - MORTY_URL=https://${SEARX_HOSTNAME}/morty/
      - MORTY_KEY=${MORTY_KEY}
    ports:
      - 127.0.0.1:8080:8080
    cap_drop:
      - ALL
    cap_add:
      - CHOWN
      - SETGID
      - SETUID
      - DAC_OVERRIDE```

Long story short, it's working for me at the moment. Please feel free to close the issue. Unless 0hlov3 has the exact same issue.

**Comment by 0hlov3** [comment]
Issue can be closed for me too. Thanks.

**Comment by dalf** [comment]
I think the different issues have been solved. Feel free to reopen.
